### PR TITLE
Test runserver

### PR DESCRIPTION
**Pull request by readevalprint**
This PR brings run the `runserver` and `runworker` management commands up to 91% and 95% respectively.  I tried to change as little in the code as possible but to capture the logging, I had to move the `handler = logging.StreamHandler()` line out of the function into the module so that it could be overridden. Without that I couldn't find a place to mock the logging.

### CODE REVIEW

File: channels/tests/test_management.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                                             host='localhost', action_logger=mock.ANY, channel_layer=mock.ANY)
+
+    @mock.patch('sys.stderr', new_callable=StringIO)
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    @mock.patch('channels.management.commands.runserver.Server')
+    def test_runserver_noworker(self, mocked_server, mock_stdout, mock_stderr):
+        '''
+        No need to Mock the Worker because it should not be used.
+        '''
+        call_command('runserver', '--noreload', '--noworker')
```
Comment (andrewgodwin) [inline channels/tests/test_management.py]: What is this testing? You seem to have mocked out the command being tested so it'll vacuously be true.

Comment (andrewgodwin) [inline channels/tests/test_management.py]: There's no assert here - what are you testing?

**Comment by andrewgodwin** [comment]
Seems fine overall, but I'd like some more comments/docstrings overall describing what's happening, as mocking this much is somewhat impenetrable to others at a first glance.

File: channels/tests/test_management.py (hunk 1 above)
Comment (readevalprint) [inline channels/tests/test_management.py]: Oh wow I just saw your comment. Good catch, let me flesh this out, not sure what I was thinking.

**Comment by readevalprint** [comment]
I'm only mocking the `sys.stdout/err` to suppress logging from the test output filling the UnitTest output. But if that is not important I can remove it. But I will add more comments regardless.

**Comment by readevalprint** [comment]
@andrewgodwin Ok, added more comments and fleshed out the assertions more. Let me know what think.

**Comment by andrewgodwin** [comment]
Sorry for the delay here - it's been a hell of a week. Looks good, thanks!

**Comment by readevalprint** [comment]
No worries, thanks!

**Pull request merged**